File: backend/src/main/java/pt/isel/rugby/model/Tournament.java
package pt.isel.rugby.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "tournament")
@Data
public class Tournament {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String name;

    @Column
    private String classification;

    @Column
    private String comment;

    @Column
    private Date date;

    @Column
    private String local;

    //TODO: hummm.
    @OneToMany(mappedBy = "tournament")
    List<Game> games;
}

File: frontend2/Rugby/src/app/classes/tournament.ts
import {Game} from '../classes/game';

export class Tournament {
  constructor(
    private id?: number,
    public name?: string,
    public date?: Date,
    public classification?: string,
    public comment?: string,
    public games?: Game[]
  ) {
    this.id = id ? id : 0;
    this.name = name ? name : '';
    this.date = date ? date : new Date(0);
    this.comment = comment ? comment : '';
    this.classification = classification ? classification : '';
    this.games = games ? games : [];
  }
}
